### PR TITLE
Mech Survivability Buff

### PR DESCRIPTION
**Pull request by Geevies**
* You can now repair mech components with nanopaste too.
* Repairing mech components now take three seconds per repair.
* Repairing mech components with welders now take a bit of fuel.
* Mechs now take much less damage than they usually did. Burning and lasers do more damage than straight brute damage.
* Shooting mechs now properly have the potential of transferring the bullet to the pilot.
* Mechs no longer die, they just go into an inactive servolock when their health goes below 0. Repair the mech to releash it from the lockdown.
* Examining a mech now properly tells you whether the legs, arms, and body are damaged to its maximum damage level. If the legs or arms are no longer functional, they will display 'Disabled' in bold red.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/heavy_vehicle/mecha.dm [hunk 1]
```diff
@@ -127,16 +128,28 @@
...
+			var/obj/item/mech_component/propulsion/P = thing
+			if(!P.motivator || !P.motivator.is_functional())
+				damage_string = SPAN_DANGER("disabled")
+		if(istype(thing, /obj/item/mech_component/manipulators))
+			var/obj/item/mech_component/manipulators/M = thing
+			if(!M.motivator || !M.motivator.is_functional())
+				damage_string = SPAN_DANGER("disabled")
+		if(istype(thing, /obj/item/mech_component/chassis))
+			if(body.total_damage == body.max_damage)
+				damage_string = SPAN_DANGER("torn open")
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/mecha.dm:142]: You can condense these two into one if/else.

File: code/modules/heavy_vehicle/components/body.dm [hunk 2]
```diff
@@ -118,4 +120,24 @@
...
 		update_components()
-	else . = ..()
\ No newline at end of file
+	else . = ..()
+
+/obj/item/mech_component/chassis/repair_brute_damage(var/amt)
+	take_brute_damage(-(amt / CHASSIS_DAMAGE_MOD))
+
+/obj/item/mech_component/chassis/repair_burn_damage(var/amt)
+	take_burn_damage(-(amt / CHASSIS_DAMAGE_MOD))
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/body.dm:129]: This is weird. You mean for this to heal 100 damage a pop? That seems excessive.

Comment (Geevies) [reply code/modules/heavy_vehicle/components/body.dm:129]: this offsets the reduction the other proc imposes

File: code/modules/heavy_vehicle/mecha.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/heavy_vehicle/mecha.dm:142]: will do tomorrow

File: code/modules/heavy_vehicle/components/body.dm (hunk 2 above)
Comment (NonQueueingMatt) [reply code/modules/heavy_vehicle/components/body.dm:129]: What reduction do you mean?

Comment (Geevies) [reply code/modules/heavy_vehicle/components/body.dm:129]: take_brute_damage and take_burn_damage multiplies by the mod to determine its amount, so it's `amt / 4` atm. so by doing the reverse on the repair one, which calls the former procs, it cancels out, giving you the regular `amt`

File: code/modules/heavy_vehicle/mech_damage.dm [hunk 3]
```diff
@@ -68,8 +72,10 @@
 	//Only 2 types of damage concern mechs and vehicles
 	switch(damagetype)
 		if(BRUTE)
+			damage *= 0.5
```
Comment (Arrow768) [inline code/modules/heavy_vehicle/mech_damage.dm:75]: Increase the component health instead.

File: code/modules/heavy_vehicle/mech_damage.dm [hunk 4]
```diff
@@ -68,8 +72,10 @@
...
 			adjustBruteLoss(damage * BLOCKED_MULT(blocked), target)
 		if(BURN)
+			damage *= 0.6
```
Comment (Arrow768) [inline code/modules/heavy_vehicle/mech_damage.dm:78]: Increase the component health instead.

File: code/modules/heavy_vehicle/mech_life.dm [hunk 5]
```diff
@@ -33,12 +33,14 @@
 		get_cell()?.drain_power(0, 0, calc_power_draw())
 
 	updatehealth()
-	if(health <= 0 && stat != DEAD)
-		death()
+	if(health <= 0 && health_lockdown == FALSE)
+		ToggleHealthLockdown()
+	else if(health > 0 && health_lockdown == TRUE)
+		ToggleHealthLockdown()
```
[the remark this answers is not in the document]
Comment (Geevies) [reply code/modules/heavy_vehicle/mech_life.dm:39]: The components inside the mech limbs still get damaged and destroyed, meaning that you have to physically take off that specific part of the mech and replace the damaged component, then re-add the part of the mech. Mid-combat repair would be suicidal.

**Comment by Geevies** [comment]
> 
> 
> Increase the health of the components and/or the effectiveness for the armor instead of adding some "magic modifiers".

Given how components are shared, like with borgs and such, tweaking all of the components would be a rework overhaul all on its own. I find that having a "magic modifier" makes it much easier for myself and anyone who comes after me to tweak how strong mechs specifically are. Additionally, the chassis determines how much health the mech has, but you don't necessarily want the mech to have a ton of health, as a four times increase to the chassis would provide. That's why I made the modifiers in the first place.

**Pull request closed without merging**